File: mq/rabbitmq/py/routing/publisher.py
#!/usr/bin/env python3
#
# Copyright (C) 2019 jayson
#
import json
import random
import logging
import utils.logging
from utils.connection import get_exchange_connection
from utils.random import random_string, random_sleep
from config import RABBITMQ_MASTER


if __name__ == '__main__':
    utils.logging.init_logging(logging.DEBUG)
    with get_exchange_connection(RABBITMQ_MASTER, 'logs_routing', 'direct') as channel:
        name = random_string(8)
        for index in range(2 ** 32):
            level = random.choice(['debug', 'info', 'warning', 'error'])
            message = json.dumps({'producer': name, 'index': index, 'from': 'python', 'body': random_string(32)})
            channel.basic_publish(exchange='logs_routing', routing_key=level, body=message.encode('utf-8'))
            utils.logging.info("send message<{} | {} | {}> to logs_routing from python".format(name, level, index))
            random_sleep(2)

File: mq/rabbitmq/php/src/routing/publisher.php
<?php
/**
 * rabbitmq hello-world
 */
require_once __DIR__ . '/../bootstrap.php';


$connection = new \PhpAmqpLib\Connection\AMQPStreamConnection(RABBITMQ_MASTER, 5672, 'guest', 'guest');
$channel = $connection->channel();

$channel->exchange_declare('logs_routing', 'direct', false, true, false);

$producer = bin2hex(random_bytes(4));
$levels = ['debug', 'info', 'warning', 'error'];
for ($index = 0; $index < 10000; ++$index) {
    $channel->basic_publish(new \PhpAmqpLib\Message\AMQPMessage(json_encode([
        'from' => 'php',
        'producer' => $producer,
        'index' => $index,
        'body' => bin2hex(random_bytes(16))
    ])), 'logs_routing', $levels[random_int(0, 3)]);
    sleep(random_int(1, 2));
}
